src/gtc/passes/oir_optimizations/utils.py

```diff
@@ -46,13 +46,14 @@ def visit_FieldAccess(
         is_write: bool,
         **kwargs: Any,
     ) -> None:
-        accesses.append(
-            Access(
-                field=node.name,
-                offset=(node.offset.i, node.offset.j, node.offset.k),
-                is_write=is_write,
+        if isinstance(node.offset.k, int):
+            accesses.append(
+                Access(
+                    field=node.name,
+                    offset=(node.offset.i, node.offset.j, node.offset.k),
+                    is_write=is_write,
+                )
             )
-        )
```

Comment:
Might be cleaner to make a `visit_VariableOffset` and `visit_CartesianOffset`, then `visit(node.offset)` here instead of checking the type of `k`.